ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/QueryCompactor.java

```diff
@@ -272,5 +277,23 @@ static void removeFilesForMmTable(HiveConf conf, AcidDirectory dir) throws IOExc
         fs.delete(dead, true);
       }
     }
+
+    static void overrideConfProps(HiveConf conf, CompactionInfo ci, Map<String, String> properties) {
+      for (String key : properties.keySet()) {
+        if (key.startsWith(COMPACTOR_PREFIX)) {
+          String property = key.substring(10); // 10 is the length of "compactor." We only keep the rest.
```

Comment:
magic number, could we use COMPACTOR_PREFIX.length() instead

Reply:
Done.